invenio_vocabularies/services/schema.py

```diff
@@ -29,10 +29,33 @@ class BaseVocabularySchema(BaseRecordSchema):
 
 
 class VocabularySchema(BaseVocabularySchema):
-    """Service schema for vocabulary records.."""
+    """Service schema for vocabulary records."""
 
     props = fields.Dict(
         allow_none=False, keys=fields.Str(), values=fields.Str()
     )
     tags = fields.List(SanitizedUnicode())
     type = fields.Str(attribute='type.id', required=True)
+
+
+class DatastreamObject(Schema):
+    """Datastream object (reader, transformer, writer)."""
+
+    type = fields.Str(required=True)
```

Comment:
Not sure if we should validate this here (kind of difficult though, since it's based on `reader/transformer/writer`) or if it's enough actually the underlying API's job to raise an appropriate exception

Reply:
I think is okay to validate up to this point (i.e. that there is a `type` and `args`, and no other field). Then, the _args_ are dependent on the `reader/transformer/writer` so that is up to them. However, with no `type` we cannot even choose one, so it is mandatory.